pkg/download/downloader.go

```diff
@@ -92,6 +105,30 @@ func extractZIP(targetDir string, read io.ReaderAt, size int64) error {
 	return nil
 }
 
+func symlink(targetDir, oldname, newname string) error {
```

Comment:
I believe we already have symlink methods in the rest of the codebase.
Would be good to extract them into a pkg perhaps.

Also this `symlink` name doesn't reflect what the method does.

Also for escape checks, can you consider using https://godoc.org/sigs.k8s.io/krew/pkg/pathutil#IsSubPath which I mentioned earlier?

Reply:
thanks for the pointer ot IsSubPath. it is close. unfortunately i think this case is a bit different. consider this case:

foo/bar/../../../baz -> newname

we need to canonicalize the oldname. go does not have a relative-canonicalize function in its standard library. there is filepath.Abs, which does canonicalize, but it also makes the path absolute.

Reply:
filepath.Clean

Reply:
Which is also used in IsSubPath :)

Reply:
thanks